### PR TITLE
Initial setup for adding Hindi to the editor.

### PR DESCRIPTION
**Pull request by abhieshekumar**
Initial setup for adding Hindi to the editor #1701 

I have verified that this pull request:

* [✓] has no linting errors (`npm run lint`)
* [✓] is from a uniquely-named feature branch and has been rebased on top of the latest `develop` branch. (If I was asked to make more changes, I have made sure to rebase onto `develop` then too)
* [✓] is descriptively named and links to an `ongoing issue #1701`

### CODE REVIEW

**Comment by catarak** [comment]
I'd prefer to merge this in with the Hindi translation done, rather than just the setup completed.

**Comment by abhieshekumar** [comment]
Sure thing. Working on that.

File: translations/locales/hi/translations.json [hunk 1]
```diff
@@ -0,0 +1,607 @@
...
+        "New": "नई",
+        "Share": "शेयर",
+        "Duplicate": "डुप्लिकेट",
+        "Open": "ओपन",
+        "Download": "डाउनलोड",
+        "AddToCollection": "संग्रह में जोड़ें",
+        "Examples": "उदाहरण"
+      },
+      "Edit": {
+        "Title": "संपादित करे ",
```
Comment (catarak) [inline translations/locales/hi/translations.json]: I think there shouldn't be a space at the end of the line here?

File: translations/locales/hi/translations.json [hunk 2]
```diff
@@ -0,0 +1,607 @@
...
+      "ErrorARIA": "एरर",
+      "Save": "सेव",
+      "p5logoARIA": "p5.js लोगो",
+      "DeleteConfirmation": "क्या आप पक्का {{name}} को डिलीट करना चाहते हैं?"
+    },
+    "IDEView": {
+      "SubmitFeedback": "प्रतिपुष्टि दें",
+      "SubmitFeedbackARIA": "प्रतिपुष्टि दें",
+      "AddCollectionTitle": "संग्रह में जोड़े",
+"AddCollectionARIA":"संग्रह में जोड़े",
```
Comment (catarak) [inline translations/locales/hi/translations.json]: would you mind fixing the indentation here?

File: translations/locales/hi/translations.json [hunk 3]
```diff
@@ -0,0 +1,607 @@
...
+        "My": "मेरा",
+        "MySketches": "मेरे स्केच",
+        "MyCollections": "मेरे संग्रह",
+        "Asset": "ऐसेट",
+        "MyAssets": "मेरे ऐसेट",
+        "LogOut": "लॉग आउट"
+      }
+    },
+    "CodemirrorFindAndReplace": {
+      "Find": "खोज",
```
Comment (catarak) [inline translations/locales/hi/translations.json:54]: You are missing a translation here (see [English JSON file](https://github.com/processing/p5.js-web-editor/blob/develop/translations/locales/en-US/translations.json#L53)): "Toggle Replace"

**Review by catarak — CHANGES_REQUESTED** [review]

**Comment by catarak** [comment]
@rt1301 I merged your changes to here! I'm going to close your other PR.

**Comment by abhieshekumar** [comment]
@catarak I have fixed the requested changes. Take a look.

**Comment by catarak** [comment]
Thank you @abhieshekumar! Everything looks good code-wise, so I'm going to solicit some feedback on this translation as you have requested 😄

File: translations/locales/hi/translations.json [hunk 4]
```diff
@@ -0,0 +1,608 @@
...
+        "FindNext": "अगला खोजें",
+        "FindPrevious": "पिछला खोजें",  
+        "Replace": "बदली करें"
+      },
+      "Sketch": {
+        "Title": "स्केच",
+        "AddFile": "फाइल जोड़ें",
+        "AddFolder": "फोल्डर जोड़ें",
+        "Run": "चलाएं",
+        "Stop": "बंद करें"
```
Comment (saRvaGnyA) [inline translations/locales/hi/translations.json]: `"Stop": "रोके"`

File: translations/locales/hi/translations.json [hunk 5]
```diff
@@ -0,0 +1,608 @@
...
+        "Share": "शेयर",
+        "Duplicate": "डुप्लिकेट",
+        "Open": "ओपन",
+        "Download": "डाउनलोड",
+        "AddToCollection": "संग्रह में जोड़ें",
+        "Examples": "उदाहरण"
+      },
+      "Edit": {
+        "Title": "संपादित करे",
+        "TidyCode": "साफ कोड",
```
Comment (saRvaGnyA) [inline translations/locales/hi/translations.json]: `"TidyCode": "कोड साफ़ करें",`

File: translations/locales/hi/translations.json [hunk 6]
```diff
@@ -0,0 +1,608 @@
...
+        "AddFile": "फाइल जोड़ें",
+        "AddFolder": "फोल्डर जोड़ें",
+        "Run": "चलाएं",
+        "Stop": "बंद करें"
+      },
+      "Help": {
+        "Title": "मदद",
+        "KeyboardShortcuts": "कीबोर्ड शॉर्टकट",
+        "Reference": "रिफरेन्स",
+        "About": "के बारे में"
```
Comment (saRvaGnyA) [inline translations/locales/hi/translations.json]: `"About": "अधिक जान्कारी"`

File: translations/locales/hi/translations.json [hunk 7]
```diff
@@ -0,0 +1,608 @@
...
+      },
+      "Lang": "भाषा",
+      "BackEditor": "एडिटर पर वापस जाएं",
+      "WarningUnsavedChanges": "क्या आप इस पेज को छोड़ना चाहते हैं? आपके पास अनसेव्ड परिवर्तन हैं।",
+      "Login": "लॉग इन",
+      "LoginOr": "या",
+      "SignUp": "साइन अप",
+      "Auth": {
+        "Welcome": "स्वागत है",
+        "Hello": "नमस्कार",
```
Comment (saRvaGnyA) [inline translations/locales/hi/translations.json]: `"Hello": "नमस्ते",`

File: translations/locales/hi/translations.json [hunk 8]
```diff
@@ -0,0 +1,608 @@
...
+      "Username": "यूजरनेम",
+      "DontHaveAccount": "अकाउंट नहीं है? ",
+      "ForgotPassword": "पासवर्ड भूल गए? ",
+      "ResetPassword": "पासवर्ड रीसेट करें"
+    },
+    "SocialAuthButton": {
+      "Connect": "कनेक्ट {{serviceauth}} अकाउंट",
+      "Unlink": "अनलिंक {{serviceauth}} अकाउंट",
+      "Login": "{{serviceauth}} से लोगिन करें",
+      "LogoARIA": "{{serviceauth}} प्रतीक चिन्ह"
```
Comment (saRvaGnyA) [inline translations/locales/hi/translations.json:89]: `      "LogoARIA": "{{serviceauth}} प्र्‌‍‌तिक‍ चिन्ह"
`

File: translations/locales/hi/translations.json [hunk 9]
```diff
@@ -0,0 +1,608 @@
...
+    },
+    "About": {
+      "Title": "के बारे में",
+      "TitleHelmet": "p5.js वेब एडिटर | के बारे में",
+      "Contribute": "योगदान",
+      "NewP5": "p5.js पर नये?",
+      "Report": "बग रिपोर्ट",
+      "Learn": "सीखें",
+      "Resources": "रीसॉर्सेज़",
```
Comment (saRvaGnyA) [inline translations/locales/hi/translations.json]: `"Resources": "साधन",`

File: translations/locales/hi/translations.json [hunk 10]
```diff
@@ -0,0 +1,608 @@
...
+      "OpenedNewSketch": "नया स्केच खोला",
+      "SketchSaved": "स्केच सेव किया",
+      "SketchFailedSave": "स्केच सेव करने में असमर्थ",
+      "AutosaveEnabled": "ऑटोसेव चालू",
+      "LangChange": "भाषा बदली",
+      "SettingsSaved": "सेटिंग्स सेव की"
+    },
+    "Toolbar": {
+      "Preview": "पूर्वावलोकन",
+      "Auto-refresh": "स्वयं रिफ़्रेश",
```
Comment (saRvaGnyA) [inline translations/locales/hi/translations.json]: `"Auto-refresh": "ऑटो-रिफ़्रेश",`

File: translations/locales/hi/translations.json [hunk 11]
```diff
@@ -0,0 +1,608 @@
...
+      "OpenPreferencesARIA": "प्राथमिकताएँ खोलें",
+      "PlaySketchARIA": "स्केच चलाएं",
+      "PlayOnlyVisualSketchARIA": "केवल दृश्य स्केच चलाएं",
+      "StopSketchARIA": "स्केच बंद करे",
+      "EditSketchARIA": "स्केच नाम संपादित करें",
```
Comment (saRvaGnyA) [inline translations/locales/hi/translations.json]: `"EditSketchARIA": "स्केच का नाम संपादित करें",`

**Comment by saRvaGnyA** [comment]
Respected Ma'am @catarak, I saw the call for review for this pull request on Processing Forums.
@abhieshekumar, terrific work on the translation buddy!

I viewed all the translations and most of them are correct. However, there are some improvements I wanted to suggest.
I've added relevant single-line comments to the required lines of the commits too. 

A major improvement I would like to suggest is wherever you have translated the word "Stop" as "बंद", please change it to "रोके". बंद literally means "to close" while रोके means "to stop".

File: translations/locales/hi/translations.json (hunk 7 above)
Comment (saRvaGnyA) [reply translations/locales/hi/translations.json]: नमस्कार is the action of greeting while नमस्ते is the greeting itself, so नमस्ते would seem more appropriate

File: translations/locales/hi/translations.json (hunk 6 above)
Comment (saRvaGnyA) [reply translations/locales/hi/translations.json]: के बारे में looks incomplete

File: translations/locales/hi/translations.json (hunk 9 above)
Comment (saRvaGnyA) [reply translations/locales/hi/translations.json]: Phonetic translation for the technical term "Resources" was not matching. Here, a translation(like in the case of the word "Example") would be more appropriate.

File: translations/locales/hi/translations.json [hunk 12]
```diff
@@ -0,0 +1,608 @@
...
+      "Sound": "साउन्ड",
+      "SoundOutputARIA": "साउन्ड आउटपुट चालू"
+    },
+    "KeyboardShortcuts": {
+      "Title": " कीबोर्ड शॉर्टकट",
+      "ShortcutsFollow": "कोड संपादन कीबोर्ड शॉर्टकट अनुसरण करते हैं",
+      "SublimeText": "Sublime Text शॉर्टकट",
+      "CodeEditing": {
+        "Tidy": "साफ",
+        "FindText": "शब्द ढूंढना",
```
Comment (saRvaGnyA) [inline translations/locales/hi/translations.json]: `"FindText": "शब्द ढूंढे",`

File: translations/locales/hi/translations.json [hunk 13]
```diff
@@ -0,0 +1,608 @@
...
+        "FindNextMatch": "अगला मिलान खोजें",
```
Comment (saRvaGnyA) [inline translations/locales/hi/translations.json:182]: `"FindNextMatch": "अगला मिलन खोजें",`

File: translations/locales/hi/translations.json [hunk 14]
```diff
@@ -0,0 +1,608 @@
...
+        "IndentCodeLeft": "इंडेंट कोड लेफ्ट",
+        "IndentCodeRight": "इंडेंट कोड राइट",
+        "CommentLine": "टिप्पणी लाइन",
+        "FindNextTextMatch": "अगला शब्द मिलान खोजें",
+        "FindPreviousTextMatch": "पिछला शब्द मिलान खोजें",
+        "CodeEditing": "कोड संपादन"
+      },
+      "General": {
+        "StartSketch": "स्केच शुरू करें",
+        "StopSketch": "स्केच बंद करो",
```
Comment (saRvaGnyA) [inline translations/locales/hi/translations.json]: `"StopSketch": "स्केच रोके",`

File: translations/locales/hi/translations.json [hunk 15]
```diff
@@ -0,0 +1,608 @@
...
+      "TableText": "टेबल-टेक्स्ट",
+      "Sound": "ध्वनि"
+    },
+    "PreferenceCreators": {
+      "On": "ऑन",
+      "Off": "औफ"
+    },
+    "MobileIDEView":{
+      "Preferences": "प्राथमिकता",
+      "MyStuff": "मेरा स्टफ",
```
Comment (saRvaGnyA) [inline translations/locales/hi/translations.json]: `"MyStuff": "मेरे वस्तु",`
मेरा स्टफ sounds pretty odd

File: translations/locales/hi/translations.json [hunk 16]
```diff
@@ -0,0 +1,608 @@
...
+      "HeaderUpdatedAt_mobile": "अपडेट किया",
+      "NoSketches": "कोई स्केच नहीं"
+    },
+    "AddToCollectionSketchList": {
+      "Title": "p5.js वेब एडिटर | मेरे स्केच",
+      "AnothersTitle": "p5.js वेब एडिटर | {{anotheruser}} के स्केच",
+      "NoCollections": "कोई संग्रह नहीं"
+    },
+    "Editor": {
+      "OpenSketchARIA": "स्केच फ़ाइलों का नेविगेशन खोलें",
```
Comment (saRvaGnyA) [inline translations/locales/hi/translations.json:530]: `"OpenSketchARIA": "स्केच फ़ाइलों का नॅविगेशन खोलें",`

File: translations/locales/hi/translations.json [hunk 17]
```diff
@@ -0,0 +1,608 @@
...
+      "CloseSketchARIA": "स्केच फ़ाइलों का नेविगेशन बंद करें",
```
Comment (saRvaGnyA) [inline translations/locales/hi/translations.json:531]: `"CloseSketchARIA": "स्केच फ़ाइलों का नॅविगेशन बंद करें",`

File: translations/locales/hi/translations.json [hunk 18]
```diff
@@ -0,0 +1,608 @@
...
+      "UnsavedChangesARIA": "स्केच में अनसेव्ड परिवर्तन हैं",
+      "KeyUpLineNumber": "लाइन {{lineNumber}}"
+    },
+    "EditorAccessibility": {
+      "NoLintMessages": "कोई लिंट मैसिज नहीं ",
```
Comment (saRvaGnyA) [inline translations/locales/hi/translations.json]: `"NoLintMessages": "कोई लिंट मैसेज नहीं ",`

File: translations/locales/hi/translations.json [hunk 19]
```diff
@@ -0,0 +1,608 @@
...
+    "SignupView": {
+      "Title": "p5.js वेब एडिटर | साइन अप",
+      "Description": "साइन अप",
+      "Or": "या",
+      "AlreadyHave": "पहले से ही अकाउंट है?",
+      "Login": "लॉग इन"
+    },
+    "EmailVerificationView": {
+      "Title": "p5.js वेब एडिटर | ईमेल वेरीफिकेशन",
+      "Verify": "अपना ईमेल वेरफाइ करें",
```
Comment (saRvaGnyA) [inline translations/locales/hi/translations.json]: `"Verify": "अपना ईमेल व्हॅरीफाय करें",`

**Comment by catarak** [comment]
Thanks for your suggestions @saRvaGnyA! @abhieshekumar you are welcome to accept these suggestions or decide they don't make sense to you, and let me know if I can support you in these decisions!

File: translations/locales/hi/translations.json (hunk 9 above)
Comment (abhieshekumar) [reply translations/locales/hi/translations.json]: I was looking for a better word for this translation. Thanks.

**Comment by abhieshekumar** [comment]
@catarak I have made the suitable changes. Thanks @saRvaGnyA.

**Comment by catarak** [comment]
Thank you for making these changes! I'm going to go ahead and merge this ✨ To be deployed soon!

**Review by catarak — APPROVED** [review]

**Pull request merged**